cmd/hub/mcscontroller/controller.go

```diff
@@ -117,16 +124,138 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 // - ServiceImports are owned by the corresponding ServiceExport and are thus
 //	 automatically deleted by k8s when their ServiceExport is deleted.
 //
-func (r *ServiceExportReconciler) reconcile(ctx context.Context, name types.NamespacedName) (ctrl.Result, error) {
-	_ = r.Log.WithValues("service", name)
+func (r *ServiceExportReconciler) reconcile(ctx context.Context, name types.NamespacedName, listOptions *client.ListOptions) (ctrl.Result, error) {
+	log := r.Log.WithValues("service", name)
+	var exports mcsv1a1.ServiceExportList
+	if err := r.Client.List(ctx, &exports, listOptions); err != nil {
+		log.Error(err, "unable to list service's service export")
+		return ctrl.Result{}, err
+	}
+	primary, err := getPrimaryExportObject(exports)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for _, se := range exports.Items {
+		if se.DeletionTimestamp != nil {
+			continue
+		}
 
-	//------------------------------------------
-	// -- @TODO CODE ABOVE SPECIFIED BEHAVIOR --
-	//------------------------------------------
+		exportSpec := &mcs.ExportSpec{}
+		err := exportSpec.UnmarshalObjectMeta(&se.ObjectMeta)
+		if err != nil {
+			log.Info("Failed to unmarshal exportSpec from serviceExport")
+			return ctrl.Result{}, err
+		}
 
+		err = exportSpec.IsCompatibleWith(primary)
+		if err != nil {
+			lhutil.UpdateServiceExportConditions(ctx, &se, log, mcsv1a1.ServiceExportConflict, corev1.ConditionTrue, "", err.Error())
+			r.Client.Status().Update(ctx, &se)
```

Comment:
error check ?

Reply:
If the exportSpecs aren't compatible it will return an error and write the error message as the reason for the conflict updated in the conditions (each error will provide a different reason for the conflict, written in the condition message). Is there something else I should add?